cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -84,6 +84,7 @@ namespace detail
 
 using shape_type = small_vector<size_t>;
 using sshape_type = small_vector<ssize_t>;
+using slice_type = small_vector<size_t>;
```

Comment:
Why do you use an unsized integer rather than a signed?  Slice can be negative-valued.

Reply:
I see, then I think your previous comment, https://github.com/solvcon/modmesh/pull/303#discussion_r1557555171, should not be an issue?

shape is always positive, and slice can be negative, so they don't need to be interchangeable.

Reply:
I misunderstood, fixed.